client/src/pages/ProfileListings/ProfileListings.tsx

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { useState, useEffect, Key } from 'react';
+import { Grid, CssBaseline, TextField, InputAdornment, Typography, Button } from '@material-ui/core';
+import SearchIcon from '@material-ui/icons/Search';
+import DateRangeIcon from '@material-ui/icons/DateRange';
+import ProfilePreview from './ProfilePreview/ProfilePreview';
+import generateMockData from './generateMockData';
+import useStyles from './useStyles';
+
+const ProfileListings = (): JSX.Element => {
+  const classes = useStyles();
+  const [currentUsers, setCurrentUsers] = useState<number>(6);
+  const [mockData, setMockData] = useState<any>([]);
+
+  const renderedPreviews = mockData.map((data: { name: any; picture: any }[], idx: Key | null | undefined) => {
+    return (
+      <Grid item key={idx}>
```

Comment:
It is not recommended to use indexes for keys. This is an anti-pattern in React.

Reply:
I just looked, the random user generator api has a nice id property to use for the key. Thanks for pointing this out!